packages/sdk/src/actions/getQuote.ts

```diff
@@ -160,24 +160,35 @@ export async function getQuote(params: GetQuoteParams): Promise<Quote> {
   // If a given cross-chain message is dependent on the outputAmount, update it
   if (crossChainMessage && typeof crossChainMessage === "object") {
     for (const action of crossChainMessage.actions) {
-      if (action.update || action.updateCallData || action.updateValue) {
-        let callData: Hex;
-        let value: bigint;
+      if (action.update || action.updateCallData || action.updateValue || action.updateAsync || action.updateCallDataAsync || action.updateValueAsync) {
```

Comment:
Are some of these fields mutually exclusive? i.e. can `action.update` be set concurrently w/ `action.updateAsync` ? The subsequent if statements seem to indicate that they could but intuitively I feel like the options should be one or the other.

Reply:
True, but I think it's better to leave this api as un-opinionated as possible, let the consumer of this api make those kinds of decisions.

Reply:
I don't think they really have that choice though, do they? It looks like the result they get is dependent on the ordering of the if statement - if they set `action.update` and `action.updateAsync` then they'll unconditionally get the async variant because the sync output would be overridden.

Just ooc, is there a problem with just forcing the function to be async? The downside I can see is that it forces non-async users (probably most use cases) to use a `Promise.resolve()` when returning. That does seem a bit annoying, tbf.

Reply:
I've implemented this and it does work, but the type is just quite complicated 😅  wdyt?